go/arrow/datatype_fixedwidth.go

```diff
@@ -88,11 +84,7 @@ func (d Date32) FormattedString() string {
 
 // Date64FromTime returns a Date64 value from a time object
 func Date64FromTime(t time.Time) Date64 {
-	if _, offset := t.Zone(); offset != 0 {
-		// properly account for timezone adjustments before we calculate
-		// the actual value by adjusting the time and converting to UTC
-		t = t.Add(time.Duration(offset) * time.Second).UTC()
-	}
+	t = t.In(time.UTC)
```

Comment:
Time.Unix() claims it is independent of the location, is this strictly needed?

Reply:
Since in theory if this change were needed, the adjustment above and the adjustment here would be the same?

Reply:
Good point,  you're absolutely right. I'll update and remove those lines